### ISSUE TITLE
or does not work well for hostname in Beaker XML

### ISSUE THREAD

**Issue by skycastlelily** [teemtee/tmt#3274]
Here is the or part of the plan,with which I try to provision a server from a certain list:
```
        or:
          - hostname: dummy1.redhat.com
          - hostname: dummy2.redhat.com
```
Here is the hostname part of the plan file:

```
        <and>
          <hostname op="==" value="dummy1.redhat.com"/>
        </and>
```

which means the job will  only try to reserve dummy1.redhat.com

**Comment by happz** [comment]
The specification looks correct, there might be a problem in how `or` - and possibly `and` as well, they are very similar - is handled by the `mrack` plugin.

**Comment by skycastlelily** [comment]
Yeah, this only affects mrack, not hostname specific, cpu.model,etc :)

**Comment by happz** [comment]
This has been fixed in https://github.com/teemtee/tmt/pull/3327
